### PR TITLE
Refine build process and output

### PR DESCRIPTION
**Pull request by Loyalsoldier**
- Detect and set $GOPATH automatically. Now it can generate `dlc.dat` without setting $GOPATH explicitly.
- Print more explicit errors
- Refine output
- Fail fast

### CODE REVIEW

**Comment by iochen** [comment]
Suggestion: you'd better not use capitalized letter in error string
Reference: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by Loyalsoldier** [comment]
@iochen Please review again.

File: main.go [hunk 1]
```diff
@@ -226,12 +230,29 @@ func main() {
...
+			output, err := exec.Command("go", "env", "GOPATH").Output()
+			if err != nil {
+				fmt.Println("Failed: please set '$GOPATH' manually, or use 'datapath' option to specify the path to your custom 'data' directory")
+				return
+			}
+			goPath = string(output)
+			goPath = strings.TrimSuffix(goPath, "\n")
+			goPath = strings.TrimSuffix(goPath, "\r")
+			fmt.Println("Use default $GOPATH set by golang:", goPath)
+			fmt.Printf("Searching directory '%s' in '%s'...\n", defaultDataPath, goPath)
```
Comment (iochen) [inline main.go:247]: default GOPATH can be read by using
```go
build.Default.GOPATH // import "go/build"
```

File: main.go [hunk 2]
```diff
@@ -226,12 +230,29 @@ func main() {
...
+		}
```
Comment (iochen) [inline main.go:248]: They can be replaced by 
```go
build.Default.GOPATH // import "go/build"
```

Reference:
https://golang.org/src/go/build/build.go#L273 => https://golang.org/src/go/build/build.go#L296

Comment (Loyalsoldier) [reply main.go:248]: I have tested it.

If you do not set `$GOPATH` environment variable explicitly, then Go runtime will use the one in `go env GOPATH` by default. In this case, if you have set `go env -w GOPATH=/a/b/c`, then `go get -u github.com/a/repo` will be installed in `/a/b/c/src/github.com/a/repo`, not the one of `build.Default.GOPATH`.

Then the build process will fail.

Comment (Loyalsoldier) [reply main.go:248]: image

Comment (iochen) [reply main.go]: It seems that `go env -w` will write environment into **EnvFile**.
So how about reading file from **EnvFile**, which can be found by a function like this https://github.com/golang/go/blob/c4f2a9788a7be04daf931ac54382fbe2cb754938/src/cmd/go/internal/cfg/cfg.go#L151 ?

I don't recommend using `exec.Command`, cause in some case, you may face some complex outside-environment problems.

Comment (Loyalsoldier) [reply main.go]: It seems that's an internal package.

Comment (iochen) [reply main.go]: yes, I mean build a function like this

**Comment by Loyalsoldier** [comment]
@iochen Please review again.

**Pull request merged**